From 751d77228aec8c8b2cc3c03bd8e7378b0bada5c3 Mon Sep 17 00:00:00 2001
From: Sergey Fedorov <barracuda@macos-powerpc.org>
Date: Sat, 22 Aug 2026 00:59:20 +0000
Subject: [PATCH 07/36] plugins: convert new-style connects to SIGNAL/SLOT
 syntax

Converts the remaining new-style connects in AbookAddressbook,
PasswordPlugin and QtKeyChainPassword; the Q_PLUGIN_METADATA/
Q_EXPORT_PLUGIN2 dual-export machinery itself was already restored
in the af84b80c ifdef revert.
---
 src/Plugins/AbookAddressbook/AbookAddressbook.cpp |  2 +-
 src/Plugins/AbookAddressbook/be-contacts.cpp      | 12 ++++++------
 src/Plugins/PasswordPlugin.cpp                    |  8 ++++----
 src/Plugins/PluginManager.h                       |  1 +
 src/Plugins/QtKeyChain/QtKeyChainPassword.cpp     |  2 +-
 5 files changed, 13 insertions(+), 12 deletions(-)

diff --git a/src/Plugins/AbookAddressbook/AbookAddressbook.cpp b/src/Plugins/AbookAddressbook/AbookAddressbook.cpp
index 62ae5498..a1448c24 100644
--- a/src/Plugins/AbookAddressbook/AbookAddressbook.cpp
+++ b/src/Plugins/AbookAddressbook/AbookAddressbook.cpp
@@ -211,7 +211,7 @@ void AbookAddressbook::scheduleAbookUpdate()
     if (!m_updateTimer) {
         m_updateTimer = new QTimer(this);
         m_updateTimer->setSingleShot(true);
-        connect(m_updateTimer, &QTimer::timeout, this, &AbookAddressbook::updateAbook);
+        connect(m_updateTimer, SIGNAL(timeout()), this, SLOT(updateAbook()));
     }
     m_updateTimer->start(500);
 }
diff --git a/src/Plugins/AbookAddressbook/be-contacts.cpp b/src/Plugins/AbookAddressbook/be-contacts.cpp
index e239fa61..6456e6e9 100644
--- a/src/Plugins/AbookAddressbook/be-contacts.cpp
+++ b/src/Plugins/AbookAddressbook/be-contacts.cpp
@@ -98,7 +98,7 @@ BE::Contacts::Contacts(AbookAddressbook *abook): m_abook(abook), m_dirty(false)
     m_sortFilterProxy->setFilterCaseSensitivity(Qt::CaseInsensitive);
     m_sortFilterProxy->setFilterKeyColumn(-1);
     m_sortFilterProxy->setSourceModel(m_abook->model());
-    connect(m_ui->filter, &QLineEdit::textChanged, m_sortFilterProxy, &QSortFilterProxyModel::setFilterWildcard);
+    connect(m_ui->filter, SIGNAL(textChanged(QString)), m_sortFilterProxy, SLOT(setFilterWildcard(QString)));
     m_ui->filter->installEventFilter(this);
 
     QFont fnt = m_ui2->name->font();
@@ -111,22 +111,22 @@ BE::Contacts::Contacts(AbookAddressbook *abook): m_abook(abook), m_dirty(false)
 
     m_ui->contacts->setModel(m_sortFilterProxy);
     m_ui->contacts->setSelectionMode(QAbstractItemView::SingleSelection);
-    connect(m_ui->contacts->selectionModel(), &QItemSelectionModel::currentChanged, this, &Contacts::setContact);
+    connect(m_ui->contacts->selectionModel(), SIGNAL(currentChanged(QModelIndex,QModelIndex)), this, SLOT(setContact(QModelIndex)));
     QModelIndex idx = m_sortFilterProxy->index(0,0);
     if (idx.isValid())
         m_ui->contacts->setCurrentIndex(idx);
     m_ui->contacts->installEventFilter(this);
 
-    connect(m_ui->add, &QAbstractButton::clicked, this, &Contacts::addContact);
-    connect(m_ui->remove, &QAbstractButton::clicked, this, &Contacts::removeCurrentContact);
-    connect(qApp, &QApplication::focusChanged, this, &Contacts::updateFocusPolicy);
+    connect(m_ui->add, SIGNAL(clicked(bool)), this, SLOT(addContact()));
+    connect(m_ui->remove, SIGNAL(clicked(bool)), this, SLOT(removeCurrentContact()));
+    connect(qApp, SIGNAL(focusChanged(QWidget*,QWidget*)), this, SLOT(updateFocusPolicy(QWidget*,QWidget*)));
 
     // cheat to correct the focuspolicies ;-)
     updateFocusPolicy(m_ui2->name, m_ui->filter);
 
     Q_FOREACH(const Field &field, fields) {
         if (QTextDocument *doc = field.label->findChild<QTextDocument*>()) {
-            connect(doc, &QTextDocument::contentsChanged, this, &Contacts::updateLabel);
+            connect(doc, SIGNAL(contentsChanged()), this, SLOT(updateLabel()));
         }
     }
 }
diff --git a/src/Plugins/PasswordPlugin.cpp b/src/Plugins/PasswordPlugin.cpp
index daaf1d05..422a2f74 100644
--- a/src/Plugins/PasswordPlugin.cpp
+++ b/src/Plugins/PasswordPlugin.cpp
@@ -29,10 +29,10 @@ namespace Plugins
 
 PasswordJob::PasswordJob(QObject *parent) : PluginJob(parent)
 {
-    connect(this, &PasswordJob::error, this, &PluginJob::finished);
-    connect(this, &PasswordJob::passwordAvailable, this, &PluginJob::finished);
-    connect(this, &PasswordJob::passwordDeleted, this, &PluginJob::finished);
-    connect(this, &PasswordJob::passwordStored, this, &PluginJob::finished);
+    connect(this, SIGNAL(error(Plugins::PasswordJob::Error,QString)), this, SLOT(finished()));
+    connect(this, SIGNAL(passwordAvailable(QString)), this, SLOT(finished()));
+    connect(this, SIGNAL(passwordDeleted()), this, SLOT(finished()));
+    connect(this, SIGNAL(passwordStored()), this, SLOT(finished()));
 }
 
 PasswordPlugin::PasswordPlugin(QObject *parent) : QObject(parent)
diff --git a/src/Plugins/PluginManager.h b/src/Plugins/PluginManager.h
index 5600c2a5..ad032322 100644
--- a/src/Plugins/PluginManager.h
+++ b/src/Plugins/PluginManager.h
@@ -24,6 +24,7 @@
 #define PLUGIN_MANAGER
 
 #include <memory>
+#include <vector>
 #include <QObject>
 #include <QString>
 #include <QStringList>
diff --git a/src/Plugins/QtKeyChain/QtKeyChainPassword.cpp b/src/Plugins/QtKeyChain/QtKeyChainPassword.cpp
index e6f072fd..3e283896 100644
--- a/src/Plugins/QtKeyChain/QtKeyChainPassword.cpp
+++ b/src/Plugins/QtKeyChain/QtKeyChainPassword.cpp
@@ -56,7 +56,7 @@ void QtKeyChainPasswordJob::doStart()
         break;
     }
     m_job->setAutoDelete(true);
-    connect(m_job, &QKeychain::Job::finished, this, &QtKeyChainPasswordJob::result);
+    connect(m_job, SIGNAL(finished(QKeychain::Job*)), this, SLOT(result()));
     m_job->start();
 }
 
-- 
2.43.0

